### PR TITLE
When main.jsbundle newer than app.jsbundle time, load main.jsbundle

### PR DESCRIPTION
**Pull request by qingfeng**
Please do not merge, this is a question I found, send pullrequest facilitate discussions

E.g:
user install 0.9.5，For example, using the react-native-keyboard event
user upgrade to 0.9.6，but, Still default read app.jsbundle, on a crash

So, I changed main.jsbundle update if the machine, the priority is read main.jsbundle

### CODE REVIEW

**Comment by qingfeng** [comment]
@msftclas done

**Comment by lostintangent** [comment]
We had actually planned to implement this behavior for the 1.1 update, but I forgot :/ the Cordova plugin already does this, so it would be great to get parity for React. Did you want to review this PR for merge?

File: CodePush.m [hunk 1]
```diff
@@ -27,13 +27,29 @@ + (NSURL *)getBundleUrl
...
+    NSComparisonResult result = [mainDate compare:appDate];
+    switch(result) {
+        case NSOrderedSame: // 一致したとき
+        case NSOrderedAscending: // date1が小さいとき
+            return [[NSURL alloc] initFileURLWithPath:packageFile];
+            
+        case NSOrderedDescending: // date1が大きいとき
+            return jsCodeLocation;
+    }
     return [[NSURL alloc] initFileURLWithPath:packageFile];
```
Comment (geof90) [inline CodePush.m]: If you do want to merge this, I would replace the switch-case with an if-else, that way we get rid of the last return statement and make the code look cleaner.

```objectivec
if (result == NSOrderedSame || result == NSOrderedAscending) {
    return [[NSURL alloc] initFileURLWithPath:packageFile];
} else {
    return jsCodeLocation;
}

**Comment by qingfeng** [comment]
@lostintangent @geof90 I want to know my idea is reasonable?

**Comment by geof90** [comment]
@qingfeng yes it is :)

**Comment by qingfeng** [comment]
@geof90 @lostintangent I changed over, please review, thanks

**Comment by lostintangent** [comment]
This change looks good to me.

@qingfeng could you check that you completed the entire workflow for the CLA? It should have added a new tag to this PR when you finished it. Thanks!

**Comment by qingfeng** [comment]
@lostintangent 

`You have successfully signed your document using DocuSign` :)

**Comment by lostintangent** [comment]
I just tested this and it works perfectly. Nice work @qingfeng! I'll merge this as soon as @geof90 signs off, and we can include this in the next release. I'll probably release a new version to NPM on Monday, since this is an important fix, and my other PR resolves a regression, so I'd like to get them out their quickly.

**Comment by geof90** [comment]
LGTM

**Comment by qingfeng** [comment]
:beers:

**Comment by lostintangent** [comment]
Hey @qingfeng, would you be interesting in doing a Skype call to discuss what you're up to? You've been so helpful so far and I'd love to usher stand what cool sudd you're doing :)

**Comment by qingfeng** [comment]
http://facebook.github.io/react-native/showcase.html
image

This is our company developed APP，Can you use email? ~ Skype, I may not be very convenient :)

**Pull request merged**